targets/TARGET_STM/stm_spi_api.c

```diff
@@ -866,15 +866,6 @@ int spi_master_write(spi_t *obj, int value)
     const int bitshift = datasize_to_transfer_bitshift(handle->Init.DataSize);
     MBED_ASSERT(bitshift >= 0);
 
-#if defined(LL_SPI_RX_FIFO_TH_HALF)
```

Comment:
this change / commit here is an optimization and is not directly related to the SPI 3 wire change. I would prefer this to be a separate Pull Request and to have a full non regression testing before accepting such change

Reply:
:heavy_check_mark: , the commit is removed from this Pull Request